File: chatgpt1_0/minify_files.py
#!/usr/bin/env python3
"""
minify_files.py - Utility to minify all files in a folder by removing non-printable characters
Usage: python minify_files.py <input_folder> [<output_folder>]
If output_folder is not specified, files will be overwritten in place
"""

import os
import sys
import re
import shutil

class FileMinifier:
    def __init__(self, input_folder, output_folder=None):
        self.input_folder = input_folder
        self.output_folder = output_folder
        self.overwrite = output_folder is None
        
        # Create output directory if specified and doesn't exist
        if not self.overwrite and not os.path.exists(self.output_folder):
            os.makedirs(self.output_folder)
    
    def minify_all_files(self):
        # Check if input folder exists
        if not os.path.exists(self.input_folder):
            print(f"Error: Input folder '{self.input_folder}' not found.")
            return False
        
        # Get all files in the input folder
        files = [f for f in os.listdir(self.input_folder) 
                if os.path.isfile(os.path.join(self.input_folder, f))]
        
        if not files:
            print(f"No files found in '{self.input_folder}'.")
            return False
        
        # Process each file
        for file in files:
            self.minify_file(os.path.join(self.input_folder, file))
        
        print(f"Successfully minified {len(files)} files.")
        return True
    
    def minify_file(self, file_path):
        try:
            # Read file content
            with open(file_path, 'r', errors='replace') as f:
                content = f.read()
            
            # Remove non-printable characters
            # This keeps only ASCII printable characters (32-126) and newlines
            minified_content = re.sub(r'[^\x20-\x7E\r\n]', '', content)
            
            # Determine output path
            if self.overwrite:
                output_path = file_path
            else:
                output_path = os.path.join(self.output_folder, os.path.basename(file_path))
            
            # Write minified content
            with open(output_path, 'w') as f:
                f.write(minified_content)
            
            print(f"Minified: {file_path} -> {output_path}")
        except Exception as e:
            print(f"Error processing file '{file_path}': {str(e)}")

def main():
    if len(sys.argv) < 2:
        print("Usage: python minify_files.py <input_folder> [<output_folder>]")
        print("If output_folder is not specified, files will be overwritten in place.")
        sys.exit(1)
    
    input_folder = sys.argv[1]
    output_folder = sys.argv[2] if len(sys.argv) > 2 else None
    
    minifier = FileMinifier(input_folder, output_folder)
    success = minifier.minify_all_files()
    
    sys.exit(0 if success else 1)

if __name__ == "__main__":
    main()

File: chatgpt1_0/chats/minified/minify_files.rb
#!/usr/bin/env ruby

# minify_files.rb - Utility to minify all files in a folder by removing non-printable characters
# Usage: ruby minify_files.rb <input_folder> [<output_folder>]
# If output_folder is not specified, files will be overwritten in place

require 'fileutils'

class FileMinifier
  attr_reader :input_folder, :output_folder, :overwrite

  def initialize(input_folder, output_folder = nil)
    @input_folder = input_folder
    @output_folder = output_folder
    @overwrite = output_folder.nil?
    
    # Create output directory if specified and doesn't exist
    if !@overwrite && !Dir.exist?(@output_folder)
      FileUtils.mkdir_p(@output_folder)
    end
  end

  def minify_all_files
    # Check if input folder exists
    unless Dir.exist?(@input_folder)
      puts "Error: Input folder '#{@input_folder}' not found."
      return false
    end

    # Get all files in the input folder
    files = Dir.glob(File.join(@input_folder, '*')).select { |f| File.file?(f) }
    
    if files.empty?
      puts "No files found in '#{@input_folder}'."
      return false
    end

    # Process each file
    files.each do |file|
      minify_file(file)
    end

    puts "Successfully minified #{files.length} files."
    true
  end

  private

  def minify_file(file_path)
    begin
      # Read file content
      content = File.read(file_path)
      
      # Remove non-printable characters
      # This keeps only ASCII printable characters (32-126) and newlines
      minified_content = content.gsub(/[^\x20-\x7E\r\n]/, '')
      
      # Determine output path
      output_path = @overwrite ? file_path : File.join(@output_folder, File.basename(file_path))
      
      # Write minified content
      File.write(output_path, minified_content)
      
      puts "Minified: #{file_path} -> #{output_path}"
    rescue => e
      puts "Error processing file '#{file_path}': #{e.message}"
    end
  end
end

# Main execution
if __FILE__ == $0
  if ARGV.empty?
    puts "Usage: ruby minify_files.rb <input_folder> [<output_folder>]"
    puts "If output_folder is not specified, files will be overwritten in place."
    exit 1
  end

  input_folder = ARGV[0]
  output_folder = ARGV[1]

  minifier = FileMinifier.new(input_folder, output_folder)
  success = minifier.minify_all_files
  
  exit(success ? 0 : 1)
end